apps/babydegen-ui/src/components/WithdrawAgentsFunds/InitiateWithdrawal.tsx

```diff
@@ -0,0 +1,190 @@
+        </Button>
+      </Flex>
+    }
+    type="error"
+    className="w-full"
+    style={{ background: COLOR.white, borderColor: COLOR.lightGrey }}
+  />
+);
+
+const FundsToWithdraw = () => {
+  const { isLoading, data } = useFunds();
+
+  return (
+    <Flex vertical gap={8}>
+      <Text type="secondary">
+        Funds to withdraw
+        <Tooltip title="Calculated by converting all active positions into USDC at current market rates. Final value may differ slightly at withdrawal.">
+          <InfoCircleOutlined style={{ marginLeft: 6 }} />
+        </Tooltip>
+      </Text>
+
+      {isLoading ? (
+        <Skeleton.Input />
+      ) : (
+        <Title level={3} className="m-0">
+          {data ? `$${data.total_value_usd.toFixed(2)} USDC` : NA}
+        </Title>
+      )}
+    </Flex>
+  );
+};
+
+type WithdrawInvestedFundsProps = {
+  isLoading: boolean;
+  onInitiateWithdrawal: (address: Address) => void;
+  withdrawalAddress: string;
+  onAddressChange: (address: Address) => void;
+};
+
+const WithdrawInvestedFunds = ({
+  isLoading,
+  onInitiateWithdrawal,
+  withdrawalAddress,
+  onAddressChange = () => null,
+}: WithdrawInvestedFundsProps) => (
+  <>
+    <Flex gap={4} vertical className="w-full">
+      <Text type="secondary">Withdrawal address</Text>
+      <Input
+        value={withdrawalAddress || ''}
+        onChange={(e) => onAddressChange(e.target.value as Address)}
+        placeholder="0x..."
+        className="w-full"
+      />
+      <Text type="secondary">
+        Ensure this is an EVM-compatible address you can access on all relevant chains.
+      </Text>
+    </Flex>
+
+    <Button
+      disabled={!withdrawalAddress}
+      loading={isLoading}
+      onClick={() => onInitiateWithdrawal(withdrawalAddress as Address)}
+      type="primary"
+      style={{ color: withdrawalAddress ? COLOR.black : undefined }}
+    >
+      Initiate withdrawal
+    </Button>
+  </>
+);
+
+export const InitiateWithdrawal = () => {
+  const { message } = App.useApp();
+  const { isLoading, initiateWithdraw, isError, data: withdrawDetails } = useWithdrawFunds();
+  const [withdrawalAddress, setWithdrawalAddress] = useState('');
+
+  const handleAddressChange = useCallback((address: Address) => {
+    setWithdrawalAddress(address);
+  }, []);
+
+  const handleInitiateWithdrawal = useCallback(() => {
+    if (!isAddress(withdrawalAddress)) {
+      message.error('Please enter a valid address.');
+      return;
+    }
+
+    initiateWithdraw(withdrawalAddress as Address);
+  }, [initiateWithdraw, message, withdrawalAddress]);
+
+  if (isError || withdrawDetails?.status === 'failed') {
+    return (
+      <WithdrawFailed
+        isLoading={isLoading}
+        href={withdrawDetails?.transaction_link}
+        onRetry={handleInitiateWithdrawal}
+      />
+    );
+  }
+
+  if (withdrawDetails?.status === 'completed') {
+    return <WithdrawSuccess href={withdrawDetails.transaction_link} />;
+  }
+
+  if (withdrawDetails?.message) {
```

Comment:
can we do this instead, otherwise it's not clear what is the message and why we use Loading component
```suggestion
  if (withdrawDetails?.status === 'pending') {
```